### PR TITLE
Addition to Authorization Server Setup page

### PR DESCRIPTION
**Pull request by firthmj**
Added comment on needed Keycloak configuration to resolve client IPs behind Apache Proxy

### CODE REVIEW

**Comment by firthmj** [comment]
Additional commits to fork add clarifications to "Node to Authorization Server Interactions" page

File: docs/Node to Authorization Server Interactions.md [hunk 1]
```diff
@@ -173,6 +173,8 @@ grant_type=client_credentials
 
 In this request the `client_assertion` is a self-signed JWT and `client_id` is the result from client registration.
 
+Details of the content of the `client_assertion` JWT can be found in [section 3 of RFC7523](https://tools.ietf.org/html/rfc7523#section-3). Note also, as mentioned in the [Authorization Server Setup page](Authorization%20Server%20Setup.md), that the Keycloak authentication server used as an example here is an OpenID Connect server, which places some additional restrictions on the JWT. In particular, the `iss` claim should also be the `client_id`, and the `jti` claim becomes mandatory. See the [OpenID Connect documentation for details and examples](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication)
```
Comment (lo-simon) [inline docs/Node to Authorization Server Interactions.md]: > In particular, the `iss` claim should also be the `client_id`,

The `iss` (issuer) claim should be the unique identifier for the Authorization Server that issued the JWT not the `client_id`, see https://datatracker.ietf.org/doc/html/rfc7523#section-3

**Review by lo-simon — CHANGES_REQUESTED** [review]

File: docs/Node to Authorization Server Interactions.md (hunk 1 above)
Comment (firthmj) [reply docs/Node to Authorization Server Interactions.md]: The JWT here is the self-signed one from the client to the authorization server. So the issuer is the client. Your link confirms that - the text is "a unique identifier for the entity that issued the JWT". RFC7523 is vague about exactly what the `iss` claim should be, beyond just a "unique identifier". But for an OpenID server, the documentation linked is explicit that it should be the `client_id`

Comment (lo-simon) [reply docs/Node to Authorization Server Interactions.md]: You are right, the self-signed JWT `iss` should be the `cliend_id`

**Review by lo-simon — APPROVED** [review]
LGTM

File: docs/Node to Authorization Server Interactions.md (hunk 1 above)
Comment (garethsb) [inline docs/Node to Authorization Server Interactions.md]:
```suggestion
Details of the content of the `client_assertion` JWT can be found in [section 3 of RFC7523](https://tools.ietf.org/html/rfc7523#section-3). Note also, as mentioned in the [Authorization Server Setup page](Authorization%20Server%20Setup.md), that the Keycloak authentication server used as an example here is an OpenID Connect server, which places some additional restrictions on the JWT. In particular, the `iss` claim should also be the `client_id`, and the `jti` claim becomes mandatory. See the [OpenID Connect documentation on Client Authentication](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication) for details and examples.
```

**Review by garethsb — APPROVED** [review]

**Pull request merged**